main.cpp

```diff
@@ -543,7 +546,8 @@ class AWS_IoT_MQTT_Test {
             Timer timer;
             timer.start();
             while (! _message_arrive_count) {
-                if (timer.read_ms() >= MQTT_RECEIVE_MESSAGE_WITH_SUBSCRIBED_TOPIC_TIMEOUT_MS) {
+                //if (timer.read_ms() >= MQTT_RECEIVE_MESSAGE_WITH_SUBSCRIBED_TOPIC_TIMEOUT_MS) {  //Ray closed
```

Comment:
`timer.read_ms()` is more refined than `(timer.elapsed_time()).count()`.
Why give up `timer.read_ms()` ?

Reply:
Build will export warning message as follows, so it suggests use “(timer.elapsed_time()).count()”,

[Warning] [main.cpp@549,27](mailto:main.cpp@549,27): 'read_ms' is deprecated: Use the Chrono-based elapsed_time method.  If integer milliseconds are needed, you can use `duration_cast<milliseconds>(elapsed_time()).count()` [since mbed-os-6.0.0] [-Wdeprecated-declarations]

Reply:
@ccli8
Other samples, resolve timer.read_ms() deprecated warning by (timer.elapsed_time()).count() ?

Reply:
Yes, including the file MQTT\MQTTmbed.h,

Reply:
1000*(timer.elapsed_time().count()) will be better friendly remember instead of duration_cast<milliseconds>...

Reply:
if we use 1000*(timer.elapsed_time().count()) instead of "chrono::duration_cast<chrono::milliseconds>(timer.elapsed_time()).count()", it fails to run as follows,

WiFi SSID:
SSID=HUAWEI-F2C9
WiFi PASSWD:Connected to the network successfully. IP address: 192.168.8.102
Opening network socket on network stack
Opens network socket on network stack OK
DNS resolution for a1fljoeglhtf61-ats.iot.us-east-2.amazonaws.com...
DNS resolution for a1fljoeglhtf61-ats.iot.us-east-2.amazonaws.com: 3.19.157.103:8883
Connecting with a1fljoeglhtf61-ats.iot.us-east-2.amazonaws.com:8883
Connects with a1fljoeglhtf61-ats.iot.us-east-2.amazonaws.com:8883 OK
Resolved MQTT client ID: A9436A2A-1FA4D35E-D8D5729B
MQTT connects failed: -1
MQTT disconnects OK

Reply:
`*1000` or `/1000`?

Reply:
*1000

Reply:
I try "(timer.elapsed_time()).count()/1000" to be OK, I will change into "(timer.elapsed_time()).count()/1000" , OK ?

Reply:
Agree.
It could avoid forum's complain: "chrono::duration_castchrono::milliseconds" not friendly remember.